Sources/Diff/CollectionViewChanges.swift

```diff
@@ -0,0 +1,225 @@
+//
+//  Copyright © 2019 Rosberry. All rights reserved.
+//
+
+public final class CollectionViewDeleteInsert<T>: CustomStringConvertible {
+
+    let item: T
+    let index: Int
+
+    public init(item: T, index: Int) {
+        self.item = item
+        self.index = index
+    }
+
+    public convenience init?(item: T?, index: Int?) {
+        if let item = item,
+            let index = index {
+            self.init(item: item, index: index)
+        }
+        else {
+            return nil
+        }
+    }
+
+    public var description: String {
+        return "\(index)"
+    }
+}
+
+public final class CollectionViewUpdate<T>: Hashable, CustomStringConvertible {
+
+    let oldItem: T
+    let newItem: T
+    let index: Int
+
+    public init(oldItem: T, newItem: T, index: Int) {
+        self.oldItem = oldItem
+        self.newItem = newItem
+        self.index = index
+    }
+
+    public convenience init?(oldItem: T?, newItem: T?, index: Int?) {
```

Comment:
It's kinda silly to make an optional init with optional parameters which will fail if any of parameters is nil.
Applies to other optional inits in this file as well.

Reply:
I agree this is not perfect solution, but it's convenient for example in this case
```
CollectionViewChange(insert: CollectionViewDeleteInsert(item: result.insert?.item.item,
                                                                    index: result.insert?.index),
                                 delete: CollectionViewDeleteInsert(item: result.delete?.item.item,
                                                                    index: result.delete?.index),
                                 update: CollectionViewUpdate(oldItem: result.replace?.oldItem.item,
                                                              newItem: result.replace?.newItem.item,
                                                              index: result.replace?.index),
                                 move: CollectionViewMove(item: result.move?.item.item,
                                                          from: result.move?.fromIndex,
                                                          to: result.move?.toIndex))
```
in `CollectionViewDeepDiff.swift`.
And choose your words my friend - It's kinda silly to communicate in that way

Reply:
No offence meant, sorry if it came across that way.